db/migrations/20200512141017_remove_urn_state.sql

```diff
@@ -0,0 +1,232 @@
+-- +goose Up
+-- SQL in this section is executed when the migration is applied.
+DROP FUNCTION IF EXISTS api.all_urn_states;
+
+DROP FUNCTION IF EXISTS api.get_urn;
+CREATE FUNCTION api.get_urn(ilk_identifier text, urn_identifier text, block_height bigint DEFAULT api.max_block()) RETURNS api.urn_snapshot
+    LANGUAGE sql STABLE STRICT
+    AS $_$
+
+SELECT *
+    FROM api.all_urns(get_urn.block_height)
+    WHERE ilk_identifier = get_urn.ilk_identifier
+    AND urn_identifier = get_urn.urn_identifier
+$_$;
+
+DROP FUNCTION IF EXISTS api.bite_event_urn;
+CREATE FUNCTION api.bite_event_urn(event api.bite_event) RETURNS api.urn_snapshot
+    LANGUAGE sql STABLE
+    AS $$
+SELECT *
+FROM api.get_urn(event.ilk_identifier, event.urn_identifier, event.block_height)
+$$;
+
+DROP FUNCTION IF EXISTS api.flip_bid_snapshot_urn;
+CREATE FUNCTION api.flip_bid_snapshot_urn(flip api.flip_bid_snapshot) RETURNS api.urn_snapshot
+       LANGUAGE sql STABLE
+       AS $$
+SELECT *
+FROM api.get_urn(
+     (SELECT identifier FROM maker.ilks WHERE ilks.id = flip.ilk_id),
```

Comment:
Hm, this is _way_ out of scope for this PR, but it seems like `api.get_flip` should be returning a `maker.flip` and we should be deleting `api.flip_bid_snapshot` - basically a story to do exactly what you're doing here for a different place where we've got a type used in queries + a trigger table

Reply:
Mostly just caught my eye because we have an `ilk_id` and `urn_id` on that type, instead of the identifiers - which seems odd. If we're going to have a custom type for queries, I would have thought we would have done the join to get the identifiers in the query and not exposed internal DB FKs

Reply:
I'm not quite sure I followed that last bit - is there an action item here besides a story about flip_bid_snapshot?

Reply:
No, sorry - kinda just thinking out loud. Will create stories for this stuff